core/store/src/migrations.rs

```diff
@@ -229,3 +232,58 @@ pub fn migrate_37_to_38(store: &Store) -> anyhow::Result<()> {
     update.commit()?;
     Ok(())
 }
+
+// Migrates the database from version 38 to 39.
+//
+// Rewrites TrieChanges to use TrieRefcountAddition and TrieRefcountSubtraction.
+// TrieRefcountSubtraction no longer stores the value behind the refcount,
+// because subtracting the refcount does not require specifying the value.
+pub fn migrate_38_to_39(store: &Store) -> anyhow::Result<()> {
+    #[derive(borsh::BorshDeserialize)]
+    struct LegacyTrieRefcountChange {
+        trie_node_or_value_hash: CryptoHash,
+        trie_node_or_value: Vec<u8>,
+        rc: std::num::NonZeroU32,
+    }
+
+    #[derive(borsh::BorshDeserialize)]
+    struct LegacyTrieChanges {
+        old_root: StateRoot,
+        new_root: StateRoot,
+        insertions: Vec<LegacyTrieRefcountChange>,
+        deletions: Vec<LegacyTrieRefcountChange>,
+    }
+
+    let mut update = store.store_update();
+    update.delete_all(DBCol::TrieChanges);
```

Comment:
Worried that combining `delete_all` on a column and writes to the same column in one update may backfire (if order of execution changes).